packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -538,24 +522,114 @@ export function checkVerifyClaimsForConfig(
       }
     }
   }
+  // TODO(POD-P2): List membership claims
+}
+
+/**
+ * Picks the smallest available circuit in this family which can handle the
+ * size parameters of a desired configuration.
+ *
+ * @param circuitReq the circuit size requirements
+ * @returns the circuit description, or undefined if no circuit can handle
+ *   the required parameters.
+ * @throws Error if there are no circuits satisfying the given requirements.
+ */
+export function circuitDescForRequirements(
+  circuitReq: GPCRequirements
+): ProtoPODGPCCircuitDesc {
+  for (const circuitDesc of ProtoPODGPC.CIRCUIT_FAMILY) {
+    if (circuitDescMeetsRequirements(circuitDesc, circuitReq)) {
+      return circuitDesc;
+    }
+  }
+
+  throw new Error(
+    `There are no circuit parameters satisfying these requirements: ${circuitReq}`
+  );
+}
+
+/**
+ * Checks whether a described circuit can meet given GPC size requirements.
+ * This will be true if each of the circuit's parameters is greater than or
```

Comment:
This sentence isn't true anymore, since the logic for tuples and lists is more complicated.